lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -187,6 +190,27 @@ data TransactionLayer k tx = TransactionLayer
         --
         -- Returns `Nothing` for ByronEra transactions.
 
+    , evaluateTransactionBalance
+        :: SealedTx
+        -> Node.ProtocolParameters
+        -> UTxO
+        -> [(TxIn, TxOut, Maybe (Hash "Datum"))] -- Extra UTxO
+        -> Maybe Node.Value
+        -- ^ Evaluate the balance of a transaction using the ledger. A valid
+        -- transaction must be balanced.
+        --
```

Comment:
As a further suggestion: if we adopt the term "balance", then perhaps we could add a comment similar to:

> The **_balance_** of a transaction is defined as:
> (value consumed by the transaction) − (value produced by the transaction)

This way, we avoid having to mention the fee, since the fee is part of the value that's "consumed" by the transaction.

Reply:
f20fa221163125e1fca037df114287fcb249907a

I'm in favour of `balance` as a term because it seems well-used in the node and ledger — including the ledger's `evaluateTransactionBalance` we make use of.

Delta seems to be a concept specific to coin-selection, and differs from balance in that it is not affected by the fee.